File: training-data/lua/perfect_numbers.lua
local function aliquot_sum(n)
  if n == 1 then return 0 end
  local sum, i = 1, 2
  while i * i <= n do
    if n % i == 0 then
      sum = sum + i
      if i ~= n // i then sum = sum + n // i end
    end
    i = i + 1
  end
  return sum
end

local function classify(n)
  local s = aliquot_sum(n)
  if s == n then return "perfect" elseif s > n then return "abundant" else return "deficient" end
end

for _, n in ipairs({ 6, 12, 28, 13, 8128 }) do print(n, classify(n)) end

File: training-data/perl/perfect_numbers.pl
use strict;
use warnings;

sub aliquot_sum {
    my $n = shift;
    return 0 if $n == 1;
    my $sum = 1;
    for (my $i = 2; $i * $i <= $n; $i++) {
        next if $n % $i;
        $sum += $i;
        $sum += $n / $i if $i != $n / $i;
    }
    return $sum;
}

sub classify {
    my $n = shift;
    return qw(perfect abundant deficient)[ (aliquot_sum($n) <=> $n) ];
}

print "$_ ", classify($_), "\n" for 6, 12, 28, 13, 8128;
